pages/api/webhooks/stripe.ts

```diff
@@ -40,35 +40,60 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
       return res.status(400).send(`Webhook signature verification failed.`);
     }
 
+    // One time payment / Initial Monthly Pro Plan Subscription
     if (event.type === 'checkout.session.completed') {
       const session = event.data.object as Stripe.Checkout.Session;
-      let userId = session.client_reference_id;
+      const userId = session.client_reference_id;
+      const email = session.customer_details?.email;
+
       const planCode = session.metadata?.plan_code;
+      const planGivingWeeks = session.metadata?.plan_giving_weeks;
       stripeSubscriptionId = session.subscription as string;
 
-      if (!userId || !planCode) {
-        throw new Error('missing User id or plan code from Stripe webhook');
+      if (!planCode && !planGivingWeeks) {
+        throw new Error(
+          'no plan code or plan giving weeks from Stripe webhook',
+        );
       }
 
-      if (
-        planCode === ONE_TIME_PRO_PLAN_FOR_1_MONTH ||
-        planCode === MONTHLY_PRO_PLAN_SUBSCRIPTION
-      ) {
-        const currentDate = dayjs().utc().toDate();
-        const proPlanExpirationDateUTC = dayjs(currentDate)
-          .add(1, 'month')
-          .toDate();
+      if (!email) {
+        throw new Error('missing Email from Stripe webhook');
+      }
 
-        // Only store expiration for one month plan
+      const proPlanExpirationDate = (() => {
+        // Takes plan_giving_weeks priority over plan_code
+        if (planGivingWeeks && typeof planGivingWeeks === 'string') {
+          const currentDate = dayjs().utc().toDate();
+          return dayjs(currentDate).add(+planGivingWeeks, 'week').toDate();
+        } else if (planCode === ONE_TIME_PRO_PLAN_FOR_1_MONTH) {
+          // Only store expiration for one month plan
+          const currentDate = dayjs().utc().toDate();
+          return dayjs(currentDate).add(1, 'month').toDate();
```

Comment:
The "Since" date, can we use the Stripe payment date instead? Because there are webhooks that is being retried at Stripe which is from weeks ago @1orZero

Reply:
Also if user has a expiration date already, can we add the planGIvingWeeks on top of that instead? Since user will purchase more later on

Reply:
DONE